### PR TITLE
Track metrics for RepaintBoundary.

### LINKED ISSUE #475: Add statistics about the net cost/benefit of each RepaintBoundary
Adding repaint boundaries to your tree is magical performance dust.  During the paint walk, we can gather data about which repaint boundaries are actually saving you anything.  We should gather that data and surface it to developers somehow.

### PR DESCRIPTION
**Pull request by Hixie**
Fixes https://github.com/flutter/flutter/issues/475

### CODE REVIEW

**Comment by abarth** [comment]
lgtm

**Comment by abarth** [comment]
How to the boundaries in the stocks app perform?

File: packages/flutter/lib/src/rendering/proxy_box.dart [hunk 1]
```diff
@@ -1480,11 +1480,113 @@ class RenderPointerListener extends RenderProxyBoxWithHitTestBehavior {
...
+    assert(() {
+      inReleaseMode = false;
+      if (debugSymmetricPaintCount + debugAsymmetricPaintCount == 0) {
+        description.add('usefulness ratio: no metrics collected yet (never painted)');
+      } else {
+        double percentage = 100.0 * debugAsymmetricPaintCount / (debugSymmetricPaintCount + debugAsymmetricPaintCount);
+        String diagnosis;
+        if (debugSymmetricPaintCount + debugAsymmetricPaintCount < 5) {
+          diagnosis = 'insufficient data to draw conclusion (less than five repaints)';
+        } else if (percentage > 90) {
```
Comment (abarth) [inline packages/flutter/lib/src/rendering/proxy_box.dart]: 90.0 ?

Comment (Hixie) [inline packages/flutter/lib/src/rendering/proxy_box.dart]: fixed

**Comment by Hixie** [comment]
The boundaries in stocks all seem to get "outstanding" if I collect enough data (98%, hundreds of redraws on the inside vs less than a dozen repaints on the other), otherwise they get "insufficient data to draw conclusion (less than five repaints)".

Obviously the really useful data point would be to figure out where a repaint boundary should be that it currently isn't, but I don't know how we'd be able to track that really. Maybe track places where paint is called when the target is not needsPaint? But that's another bug.

**Pull request merged**